Runtime information

hjanuschka opened the pull request:
as discussed here: https://github.com/getsentry/sentry-php/issues/562
this one adds a context containing the current php version.

running the `examples/vanilla/index.php`  results in a exception like this:
( i wanted to upload via github (seems to 503 right now, hope they have sentry))

https://imgur.com/Hs6FDYm



i really would appreciate if we can get this in an 1.8.* release (atleast a release in a short time) - i really would love to have this in production. let me know if you need me to backport it somewhere.

ste93cry commented on lib/Raven/Client.php:
```diff
@@ -884,6 +884,8 @@ public function capture($data, $stack = null, $vars = null)
         if (empty($data['site'])) {
             unset($data['site']);
         }
+        $data['contexts']['runtime'] =  array('version' => phpversion(), 'name' => 'php');
```
The blank line added below this line should probably go above. Also, maybe it would be better to have the `php` string in uppercase. Last thing, the most important: please use the `PHP_VERSION` constant as it's faster because we don't have the overhead of a function call

dcramer replied:
'php' should not be uppercased as this is a value sent to the Sentry server and is reflective on the actual platform (it generally should always be lowercased)

Jean85 replied:
That string is strictly validated on server side, so we cannot change it, see [the docs about this](https://docs.sentry.io/clientdev/attributes/#required-attributes) and the related [API JSON schema](https://sentry.io/api/store/schema). 
Lowercase it is then.

hjanuschka commented:
@ste93cry feedback adressed.

ste93cry commented:
Thank you for the good work, patch looks good for me

stayallive commented on lib/Raven/Client.php:
```diff
@@ -885,6 +885,8 @@ public function capture($data, $stack = null, $vars = null)
             unset($data['site']);
         }
 
+        $data['contexts']['runtime'] =  array('version' => PHP_VERSION, 'name' => 'php');
```
Shouldn't this be merged with a possible user value in the runtime context?

Also there is an extra space character before `array`.

stayallive commented on test/Raven/Tests/ClientTest.php:
```diff
@@ -1035,6 +1035,20 @@ public function testCaptureMessageWithUserContext()
+        $events = $client->getSentEvents();
+        $this->assertEquals(PHP_VERSION, $events[0]['contexts']['runtime']['version']);
+        $event = array_pop($events);
+        $this->assertEquals('php', $event['contexts']['runtime']['name']);
```
Maybe also add an test what happens if the context was set by an user (for the runtime) it correctly merges with our runtime context.

hjanuschka commented:
@stayallive feedback addressd with: https://github.com/getsentry/sentry-php/pull/564/commits/59909d7c1c2ceb08c3327f3508c5d1be4528dc5a

verifies that custom `contextes` and `runtime` get merged

hjanuschka commented:
@Jean85 thx! - added a test to verify that on can override `[runtime][name]`  and `[runtime][version]`

had to change the order of the `array_merge` :D
also fixed tests on 5.3 :rage3:  (`[]`  vs `array()`)

hjanuschka commented:
thx, you folks rock! 🍺

hjanuschka commented:
you want me to port it into 1.8 branch ? really would love to see this released.

Jean85 commented:
@hjanuschka that's not necessary, the master branch is still for the 1.x series. The porting to that branch is done when tagging and releasing.
